[sdl2/2.0.10] Fail to install if shared + MinGW + Windows 7 without symbolic link creation permission

### Package and Environment Details
  * Package Name/Version: **sdl2/2.0.10**
  * Operating System+version: **Windows 7 (without admin rights)**
  * Compiler+version: **gcc 8.2**
  * Conan version: **conan 1.20.4**
  * Python version: **Python 3.5.4**

### Conan profile
```
[settings]
os=Windows
os_build=Windows
arch=x86_64
arch_build=x86_64
compiler=gcc
compiler.version=8
compiler.libcxx=libstdc++11
build_type=Release
[options]
[build_requires]
[env]
```
### Steps to reproduce

`conan install sdl2/2.0.10@bincrafters/stable -o sdl2:shared=True -b`

### Logs

```
CMake Error: failed to create symbolic link 'libSDL2.dll': operation not permitted
CMake Error at source_subfolder/cmake_install.cmake:176 (file):
  file INSTALL cannot find
  "C:/Users/user/.conan/data/sdl2/2.0.10/bincrafters/stable/build/e69c13a552b8475bf6ba93f2669fb026d8d4924
0/build_subfolder/lib/libSDL2.dll".
Call Stack (most recent call first):
  cmake_install.cmake:37 (include)


mingw32-make.exe: *** [Makefile:73: install] Error 1
ERROR: sdl2/2.0.10@bincrafters/stable: Error in package() method, line 288
        cmake.install(build_dir=self._build_subfolder)
        ConanException: Error 2 while executing cmake --build build_subfolder --target install -- -j4
```

--- a/conanfile.py
+++ b/conanfile.py
@@ -186,7 +186,7 @@ class SDL2Conan(ConanFile):
                               "if(NOT WINDOWS OR CYGWIN OR MINGW)")
         tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
                               "if(NOT (WINDOWS OR CYGWIN))",
-                              "if(NOT WINDOWS OR CYGWIN OR MINGW)")
+                              "if(NOT (WINDOWS OR CYGWIN OR MINGW))")
         self._build_cmake()
 
     def _check_pkg_config(self, option, package_name):
@@ -278,6 +278,8 @@ class SDL2Conan(ConanFile):
         self.output.info('Creating SDL_CONFIG environment variable: %s' % sdl2_config)
         self.env_info.SDL_CONFIG = sdl2_config
         self.cpp_info.libs = [lib for lib in tools.collect_libs(self) if '2.0' not in lib]
+        # ensure that libSDL2main is linked first
+        self.cpp_info.libs.reverse()
         if not self.options.sdl2main:
             self.cpp_info.libs = [lib for lib in self.cpp_info.libs if 'main' not in lib]
         self.cpp_info.includedirs.append(os.path.join('include', 'SDL2'))

--- a/conanfile.py
+++ b/conanfile.py
@@ -278,10 +278,11 @@ class SDL2Conan(ConanFile):
         self.output.info('Creating SDL_CONFIG environment variable: %s' % sdl2_config)
         self.env_info.SDL_CONFIG = sdl2_config
         self.cpp_info.libs = [lib for lib in tools.collect_libs(self) if '2.0' not in lib]
-        # ensure that libSDL2main is linked first
-        self.cpp_info.libs.reverse()
         if not self.options.sdl2main:
             self.cpp_info.libs = [lib for lib in self.cpp_info.libs if 'main' not in lib]
+        else:
+            # ensure that SDL2main is linked first
+            self.cpp_info.libs.insert(0, self.cpp_info.libs.pop(self.cpp_info.libs.index("SDL2main")))
         self.cpp_info.includedirs.append(os.path.join('include', 'SDL2'))
         if self.settings.os == "Linux":
             self.cpp_info.system_libs.extend(['dl', 'rt', 'pthread'])

